.github/workflows/build-web.yml

```diff
@@ -0,0 +1,80 @@
+name: 🕸️ Build (Web)
+
+on:
+  push:
+    branches:
+      - '**'
+    paths-ignore:
+      - 'docs/**'
+      - 'examples/**'
+      - '**.md'
+      - '**.txt'
+      - '!**/CMakeLists.txt'
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - 'examples/**'
+      - '**.md'
+      - '**.txt'
+      - '!**/CMakeLists.txt'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  EMSCRIPTEN_VERSION: 3.1.39
+  EMSCRIPTEN_CACHE_FOLDER: emsdk-cache
+
+jobs:
+  setup_emscripten:
+    name: Pre-warm cache for emscripten
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Set up emsdk
+        uses: mymindstorm/setup-emsdk@v14
+        with:
+          version: ${{ env.EMSCRIPTEN_VERSION }}
+          actions-cache-folder: ${{ env.EMSCRIPTEN_CACHE_FOLDER }}
```

Comment:
I'm not sure this does what you want it to do. There's no dependency set up between this job and the `build` job, so all the build jobs can/will run before the pre-warming has run.

I suspect you'll need `needs: setup_emscripten` in the `build` job.

Reply:
Indeed. Will fix. Thanks!